f5/bigip/sys/folder.py

```diff
@@ -82,3 +85,32 @@ def update(self, **kwargs):
         if inherit_device_group == 'true':
             self.__dict__.pop('deviceGroup')
         return self._update(**kwargs)
+
+    def exists(self, **kwargs):
+        """Check for the existence of the named object on the BigIP
+
+        Tries to `load()` the object and if it fails checks the exception
+        for 404.  If the `load()` is successful it returns `True` if the
+        exception is :exc:`requests.HTTPError` and the
+        ``status_code`` is ``404``
+        it will return error.  All other errors are raised as is.
+
+        :param kwargs: Keyword arguments required to load objects
+        :returns: bool -- The objects exists on BigIP or not.
+        :raises: :exc:`requests.HTTPError`, Any HTTP error that was not status
+            code 404.
+        """
+        requests_params = self._handle_requests_params(kwargs)
+        self._check_load_parameters(**kwargs)
+        load_uri = self._create_subpath_uri(kwargs)
+        session = self._meta_data['bigip']._meta_data['icr_session']
+        kwargs.update(requests_params)
+        try:
+            session.get(load_uri, **kwargs)
+        except HTTPError as err:
+            print(err.response.text)
+            if err.response.status_code == 404:
+                return False
+            else:
+                raise
+        return True
```

Comment:
No, we discussed it, and we don't want Users to depend on "exists" for object state manipulation.